code/game/objects/items/nemesis_items.dm

```diff
@@ -0,0 +1,458 @@
+	. = ..()
+	icon_state = "trap_small"
+
+/obj/item/nemesis_mine/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	if(!..()) //If not caught, deploy
+		if(overloaded)
+			trigger_overload()
+			return
+		active = TRUE
+		icon_state = "trap_small"
+		update_icon()
+		anchored = TRUE
+		playsound(src, 'sound/magic/blind.ogg', 50, TRUE)
+		if(ishuman(throwingdatum.thrower))
+			var/mob/living/carbon/human/owner = throwingdatum.thrower
+			if(istype(owner.get_item_by_slot(ITEM_SLOT_EYES), /obj/item/clothing/glasses/hud/security/sunglasses/nemesis))
+				glasses = owner.get_item_by_slot(ITEM_SLOT_EYES)
+
+/obj/item/nemesis_mine/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_SELF)
+		return
+
+	if(prob(35))
+		trigger()
+
+	var/datum/effect_system/spark_spread/sparks = new(get_turf(src))
+	sparks.set_up(4, 0, loc)
+	sparks.start()
+	qdel(src)
+
+/obj/effect/temp_visual/nemesis_circle
+	name = "energy blast"
+	icon = 'icons/obj/nemesis.dmi'
+	icon_state = "trap_explosion"
+	duration = 8
+
+/obj/effect/temp_visual/nemesis_circle/huge
+	name = "big energy blast"
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "nemesis_circle"
+	duration = 5
+	pixel_x = -32
+	pixel_x = -32
+
+/obj/item/shield/energy/nemesis
+	name = "tactical energy shield"
+	desc = "A tactical wrist-mounted energy shield from Nemesis Solutions."
+	icon = 'icons/obj/nemesis.dmi'
+	icon_state = "nemesis0"
+	base_icon_state = "nemesis"
+	force = 3
+
+	item_flags = SLOWS_WHILE_IN_HAND
+	slowdown = 0
+
+/obj/item/shield/energy/nemesis/Initialize()
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, CLOTHING_TRAIT)
+
+/obj/item/shield/energy/nemesis/attack_self(mob/living/carbon/human/user)
+	if(clumsy_check && HAS_TRAIT(user, TRAIT_CLUMSY) && prob(50))
+		to_chat(user, "<span class='userdanger'>You beat yourself in the head with [src]!</span>")
+		user.take_bodypart_damage(5)
+
+	var/obj/item/clothing/gloves/rapid/nemesis/gloves = user.get_item_by_slot(ITEM_SLOT_GLOVES)
+	if(!gloves || !istype(gloves))
+		return FALSE
+
+	if(!gloves.charge)
+		to_chat(user, "<span class='warning'>[src] fails to activate without any charge avalible.</span>")
+		return FALSE
+
+	active = !active
+	icon_state = "[base_icon_state][active]"
+
+	if(active)
+		force = on_force
+		throwforce = on_throwforce
+		throw_speed = on_throw_speed
+		w_class = WEIGHT_CLASS_BULKY
+		playsound(user, 'sound/weapons/saberon.ogg', 35, TRUE)
+		to_chat(user, "<span class='notice'>[src] is now active.</span>")
+		slowdown = 2 //Heavy slowdown
+	else
+		force = initial(force)
+		throwforce = initial(throwforce)
+		throw_speed = initial(throw_speed)
+		w_class = WEIGHT_CLASS_TINY
+		playsound(user, 'sound/weapons/saberoff.ogg', 35, TRUE)
+		to_chat(user, "<span class='notice'>[src] can now be concealed.</span>")
+		slowdown = 0
+	add_fingerprint(user)
+
+/obj/item/shield/energy/nemesis/on_shield_block(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", damage = 0, attack_type = MELEE_ATTACK)
```

Comment:
does this run if you're attacked in melee too? or just when the shield SUCCESSFULLY reflects something?

Reply:
wait, hold on, because hit_reaction just returns 0 for energy shields (and thus never calls on_shield_block()), I think that this on_shield_block() code never actually runs for energy shields (and their subtypes).

Reply:
Thanks, I should delete this